### DISCUSSION TITLE
Pixelfed `Reading Exif data is not supported by this PHP installation` error uploading story

### DISCUSSION THREAD

**Discussion by fishcharlie** [pixelfed/pixelfed#5073] [Q&A]
I recently created a Pixelfed server using Docker (https://blog.pixelfed.de/2020/05/29/pixelfed-in-docker/), and when trying to upload a story I get an error: `Reading Exif data is not supported by this PHP installation`.

But using the Vernissage app to upload a standard post it worked fine.

Any ideas how I could begin even trying to debug this? Haven't used PHP in a while.

**Comment by nhurel** [comment]
Hello,
I've had the same issue. I've tried installing/enabling missing php extensions (mbstring, exif) but never succeed. In the end, I have made a copy of the entrypoint.sh of the worker container and added `apt update && apt install -y imagemagick` at the beginning of the script. Then I set the environment variable `IMAGE_DRIVER=imagick`  in the `.env` file (instead of `gd`) and now it works !

It's a crappy workaround. Hopefully they will fix the root issue in the docker image :crossed_fingers:

**Reply by fishcharlie** [reply]
Is the `entrypoint.sh` file you made in this repo itself? Not finding that file anywhere in the Docker volume.

**Reply by nhurel** [reply]
The repository building the docker image referenced in https://blog.pixelfed.de/2020/05/29/pixelfed-in-docker/ is hosted there : https://git.zknt.org/dockers/pixelfed.
So you need to copy this file https://git.zknt.org/dockers/pixelfed/src/branch/master/websockets-entrypoint.sh and add the following on line 3 : `apt update && apt install -y imagemagick `

Then you can overwrite the entrypoint file by adding this volume in the docker-compose for the worker service : 
`./worker-entrypoint.sh:/worker-entrypoint.sh`

HTH
